### ISSUE TITLE
Check for proper nesting

### ISSUE THREAD

**Issue by kevincolten** [htmllint/htmllint#164]
Is there a way to lint for proper nesting of elements?

**Comment by mlochbaum** [comment]
Improper nesting is invalid html, so it will fail to parse correctly. This usually causes a "tag is not closed" error, provided `tag-close` is set, which is true by default. However, htmllint's output is unlikely to be as good as that of a dedicated validator.

If you know of any cases for which htmllint doesn't give an error on improper nesting, or gives a confusing error, let me know and I will be happy to take a look at it.

**Comment by rajsite** [comment]
My installed version (htmllint-cli 0.0.5 which installed htmllint 0.3.0) and the demo on the homepage both do not show an error in the following example:
```html
<!DOCTYPE html>
<html>
<head>
<title>An example page</title>
</head>
<body>
<main>
<leaflet-map-zoom-control></leaflet-map-zoom>
</main>
</body>
</html>
```

Setting the "tag-close" property to true, false or not including the property does not seem to have an effect

**Comment by mlochbaum** [comment]
Your example lints correctly (yielding a "tag is not closed" error) in version 0.5.0. I will be updating htmlling-cli this week. I'm not really sure how the homepage demo works but I will plan on looking at that as well.

**Comment by mlochbaum** [comment]
#176 provides a pretty good list of ways in which htmllint fails on improperly-nested elements, so I'm closing this issue.
